dask/utils.py

```diff
@@ -893,6 +893,10 @@ def is_arraylike(x):
     >>> is_arraylike('cat')
     False
     """
-    return (hasattr(x, '__array__') and
+    import dask.dataframe as dd
+    if isinstance(x, (dd.DataFrame, dd.Series)):
+        return False
+    else:
+        return (hasattr(x, '__array__') and
```

Comment:
There are a couple problems with this:

1.  We can't assume that dask.dataframe will import cleanly here.  Dask doesn't require users be able to install dask dataframe in order to use other systems, like dask.bag or dask.array
2.  Arguably `dd.Series` is array like by this definition, no?

Perhaps instead `.shape` should return a tuple of delayed objects rather than a delayed tuple?

Reply:
+1 on returning a tuple of delayed objects rather than a delayed tuple.

Reply:
I agree with both of you and that was the first approach that I tried. The "problem" with a tuple of delayed objects is that we need to create a class dask.tuple, otherwise, tuple don't have the compute method and create it, maybe, is a bit overkill.

Somewhere on dask documentation, I saw a scratch of dask.tuple implementation. Do you know if it's usable?

Reply:
The top-level `dask.compute()` function is probably the best way to evaluate these.

I suppose the other option is to return a 1D dask array instead of a tuple.

Reply:
I think we should avoid that idea in order to be coherent with Pandas API.

Reply:
> ...the other option is to return a 1D dask array instead of a tuple.

This could also be turned into a `tuple` of `int`s once the 1-D Dask Array is created.